### ISSUE TITLE
How does the dual licensing work for commercial apps based on PageTurner?

### ISSUE THREAD

**Issue by mmrasheed** [NightWhistler/PageTurner#586]
Hi,
Most of the codes of PageTurner is GPL-ed, where as the contributed codes are supposed to be Apache licensed. Please clarify how it works for closed source apps based on PageTurner source code. Does it have to open source modifications? Or only the base source code from PageTurner? 

Best regards,
 M.

**Comment by NightWhistler** [comment]
There are two ways to create a derived app based on the PageTurner source code:

  - Use the code for free, and license your app under the GPL. This means that you are required to publish the full source code of your app, so that will automatically include all the changes you have made to the PageTurner source code. This also means your app can never be closed source.
  - Buy a commercial license. If you buy a commercial license from Digital Trees, you are no longer required to publish any source code. This means you can keep your app closed source.

The reasoning behind all of this is: I own the copyright to the main PageTurner code base, so I can publish it under both a commercial license and the GPL license. Contributions by others, to which I don't own the copyright are automatically licensed to me under the Apache license which allows me to sublicense them as part of a commercial offering.

I hope this answers your question.

In summary, you can only create a closed source app based on PageTurner if you buy a commercial license. In all other cases, you are required to publish the source code of your derived app.

**Comment by mmrasheed** [comment]
Thanks for the clarification. May I inquire the pricing of commercial license in details?

**Comment by NightWhistler** [comment]
Please send e-mail to info@pageturner-reader.org
